backend/ticketvise/views/api/ticket.py

```diff
@@ -185,15 +186,32 @@ def get(self, request, inbox_id):
         labels = list(map(int, request.GET.getlist("labels[]", [])))
 
         inbox = get_object_or_404(Inbox, pk=inbox_id)
-        tickets = Ticket.objects.filter(inbox=inbox, title__icontains=q) | \
-                  Ticket.objects.filter(inbox=inbox, ticket_inbox_id__icontains=q).order_by("-date_created")
-
-        for term in q.split():
-            tickets = tickets | \
-                      (Ticket.objects.filter(author__username__icontains=term) |
-                       Ticket.objects.filter(author__first_name__icontains=term) |
-                       Ticket.objects.filter(author__last_name__icontains=term) |
-                       Ticket.objects.filter(author__email__contains=term))
+        # Only search if q for improved speed.
+        if q:
+            query = SearchQuery(q)
+
+            # Load comments and replies if permissions are right, else load only replies.
+            is_reply = [True]
+            if request.user.is_assistant_or_coordinator(inbox) or request.user.is_superuser:
+                is_reply = [True, False]
+
+            users = User.objects.annotate(search=SearchVector("first_name", "username", "last_name", "email")).filter(
```

Comment:
I believe this will not work similar to the previous setup, try searching for `<some_email>@`, note the use of the special character (`@` is not the only one).

Reply:
Searching for the email works, as long as the complete email is used.